Updated asset types page with all asset types supported in Automated Testing project (#1687)
o3de/o3de.org: content/docs/user-guide/assets/asset-types.md

toc: true
---

> The table below describes some of the most common source asset types supported by the **Asset Pipeline** in **Open 3D Engine (O3DE)** and lists the product assets generated by **Asset Processor** for each source asset type. The table also lists which builders process which source asset types, and the location of those builders. Builders can be located in the O3DE engine code, Gems, or project code.

Author: Fixed

{{< note >}}
Some source assets are simply copied to the **Asset Cache**, so the product asset type is identical to the source asset type.
{{< /note >}}

## Asset Builders

| Source Asset | Builder | Builder Location | Description  | Product Assets | 
| - | - | - | - | - |
| `.animgraph` | AnimGraphBuilderWorker | [EMotionFX Gem](https://github.com/o3de/o3de/tree/development/Gems/EMotionFX) | [Animation graphs](docs/user-guide/visualization/animation/character-editor/concepts-and-terms) define animation behavior for characters. | `.animgraph` |

> | `.attimage`, `.azasset`, `.azbuffer` | Any Asset Builder | [Atom Gem](https://github.com/o3de/o3de/blob/development/Gems/Atom/RPI/Code/Source/RPI.Builders/Common/AnyAssetBuilder.h) | - | Same as input, `.attimage` generates an `.attimage` product asset. |

Author: Fixed

| `.texatlas` | Atlas Worker Builder | [Texture Atlas Gem](https://github.com/o3de/o3de/blob/development/Gems/TextureAtlas/Code/Source/Editor/AtlasBuilderWorker.h) | [Texture atlases](docs/user-guide/interactivity/user-interface/canvases/texture-atlases) are used to reduce draw calls in user interfaces. | `.texatlasidx` and `.streamingimage` |

> | `.bmp`, `.dds`, `.exr`, `.gif`, `.jpeg`, `.jpg`, `.png`, `.tga`, `.tiff`, `.tif` | Atom Image Builder | [Atom Gem](https://github.com/o3de/o3de/blob/development/Gems/Atom/Asset/ImageProcessingAtom/Code/Source/ImageBuilderComponent.h) | Processes images for use in O3DE, primarily for use as [Textures](docs/user-guide/assets/texture-settings/texture-assets). | Output depends on settings, but will generally include `.imagemipchain` and `.streamingimage` |

Author: Fixed, with a slight change to "Product assets generated"

| `.materialtype`, `.material` | Atom Material Builder | [Atom Gem](https://github.com/o3de/o3de/blob/development/Gems/Atom/RPI/Code/Source/RPI.Builders/Material/MaterialBuilder.h) | [Materials](docs/atom-guide/look-dev/materials) contain properties to define how objects are rendered. | `.materialtype` outputs `.material` and `.azmaterialtype` products, `.material` outputs `.azmaterial` products. |

> | `.resourcepool` | Atom Resource Pool Asset Builder | [Atom Gem](https://github.com/o3de/o3de/blob/development/Gems/Atom/RPI/Code/Source/RPI.Builders/ResourcePool/ResourcePoolBuilder.h) | - | `.pool`, or `.streamingimagepool` |

Author: Fixed

| `.benchmarksettings`, `.benchmark` | Benchmark Asset Worker Builder | [LmbrCentral Gem](https://github.com/o3de/o3de/blob/development/Gems/LmbrCentral/Code/Source/Builders/BenchmarkAssetBuilder/BenchmarkAssetBuilderComponent.h) | - | - |
| `.blast` | Blast Scene Builder | [Blast Gem](https://github.com/o3de/o3de/blob/development/Gems/Blast/Editor/Scripts/blast_asset_builder.py) | - | `.blast`, `.assetinfo.generated` |
| `.cfg` | CFG Builder Worker | [LmbrCentral Gem](https://github.com/o3de/o3de/blob/development/Gems/LmbrCentral/Code/Source/Builders/CopyDependencyBuilder/CfgBuilderWorker/CfgBuilderWorker.h) | - | `.cfg` |
| `.emfxworkspace` | EmfxWorkspaceBuilderDescriptor | [LmbrCentral Gem](https://github.com/o3de/o3de/blob/development/Gems/LmbrCentral/Code/Source/Builders/CopyDependencyBuilder/EmfxWorkspaceBuilderWorker/EmfxWorkspaceBuilderWorker.h) | - | `.emfxworkspace` |

> | `.fontfamily`, `.font` | FontBuilderWorker | [LmbrCentral Gem](https://github.com/o3de/o3de/blob/development/Gems/LmbrCentral/Code/Source/Builders/CopyDependencyBuilder/FontBuilderWorker/FontBuilderWorker.h) | [Fonts](docs/user-guide/interactivity/user-interface/fonts) are used to render text. | Same as input, `.font` generates a `.font` product asset. |

Author: Fixed

| `.tfx` | HairAssetBuilder | [AtomTressFX Gem](https://github.com/o3de/o3de/blob/development/Gems/AtomTressFX/Code/Builders/HairAssetBuilder.h) | [TressFX](docs/user-guide/gems/reference/rendering/amd/atom-tressfx) provides realistic hair and fur simulation. | `.tfxhair` |
| `.lua` | Lua Worker Builder | [LmbrCentral Gem](https://github.com/o3de/o3de/blob/development/Gems/LmbrCentral/Code/Source/Builders/LuaBuilder/LuaBuilderWorker.h) | [Lua](docs/user-guide/scripting/lua) is a scripting language that can be used to author logic for your project. | `.luac` |

> | `.mock` | Mock Builder | [Automated Testing Project](https://github.com/o3de/o3de/blob/development/AutomatedTesting/Gem/PythonTests/PythonAssetBuilder/mock_asset_builder.py) | This builder is used to assist in some automated testing, run via the AutomatedTesting project. It's a useful example for teams that wish to author builders for custom asset types in their own project, in Python. | `.mock_asset` |

Author: Fixed, with an additional change to remove the "is" before assists

| `.motionset` | MotionSetBuilderWorker | [EmotionFX Gem](https://github.com/o3de/o3de/blob/development/Gems/EMotionFX/Code/EMotionFX/Pipeline/EMotionFXBuilder/MotionSetBuilderWorker.h) | A [motion set](docs/user-guide/visualization/animation/character-editor/concepts-and-terms) contains a list of motions. | `.motionset` |

> | `.pass` | Pass Asset Builder | [Atom Gem](https://github.com/o3de/o3de/blob/development/Gems/Atom/RPI/Code/Source/RPI.Builders/Pass/PassBuilder.h) | [Passes](docs/atom-guide/dev-guide/passes) are logical groupings of render work, with a defined input and output. | `.pass` |

Author: Fixed

| `.precompiledshader` | Precompiled Shader Builder | [Atom Gem](https://github.com/o3de/o3de/blob/development/Gems/Atom/Asset/Shader/Code/Source/Editor/AzslShaderBuilderSystemComponent.h) | - | `.azshader` |
| `.prefab` | Prefab Builder | [Prefab Gem](https://github.com/o3de/o3de/blob/development/Gems/Prefab/PrefabBuilder/PrefabBuilderComponent.h) | [Prefabs](docs/user-guide/interactivity/prefabs) are a foundational element for building an O3DE project, a prefab is a collection of entities. | `.spawnable` |

> | `.ts` | Qt Translation File Builder | [LmbrCentral Gem](https://github.com/o3de/o3de/blob/development/Gems/LmbrCentral/Code/Source/Builders/TranslationBuilder/TranslationBuilderComponent.h) | Qt translation files. Qt is only enabled in tools, so these can be useful for translating tools to different languages, but not projects. | `.qm` |

Author: We really need to clean up some terminology. There are non-Editor tools, like Asset Processor and Asset Bundler, that can use the translation files. I've changed it to your suggestion for now because I think we need to do an entire-documentation pass on the terminology.

> | `.ts` | Qt Translation File Builder | [LmbrCentral Gem](https://github.com/o3de/o3de/blob/development/Gems/LmbrCentral/Code/Source/Builders/TranslationBuilder/TranslationBuilderComponent.h) | Qt translation files. Qt is only enabled in tools, so these can be useful for translating tools to different languages, but not projects. | `.qm` |

Reviewer: I'll pass this feedback on to the rest of the team.

| `.xmlschema` | SchemaBuilderWorker | [LmbrCentral Gem](https://github.com/o3de/o3de/blob/development/Gems/LmbrCentral/Code/Source/Builders/CopyDependencyBuilder/SchemaBuilderWorker/SchemaBuilderWorker.h) | XML Schema files are used to assist in finding product dependencies in XML files. The XML Schema Editing tool in the O3DE Editor can define patterns to match XML format files, and patterns search for product dependencies in those files. | `.xmlschema` |
| `.scriptcanvas` | Script Canvas Builder | [ScriptCanvas Gem](https://github.com/o3de/o3de/blob/development/Gems/ScriptCanvas/Code/Builder/ScriptCanvasBuilderWorker.h) | [Script Canvas](docs/user-guide/scripting/script-canvas) is a general purpose, visual scripting environment for O3DE. | `.luac`, `.scriptcanvas_compiled`, `.scriptcanvas_fn_compiled` |

> | `.scriptevents` | Script Events Builder | [ScriptEvents Gem](https://github.com/o3de/o3de/blob/development/Gems/ScriptEvents/Code/Builder/ScriptEventsBuilderWorker.h) | [Script Events](docs/user-guide/scripting/script-events) are used to allow scripts to communicate with each other. | `.scriptevents` |

Author: Fixed

| `engine.json` | Settings Registry Builder | [Asset Processor](https://github.com/o3de/o3de/blob/development/Code/Tools/AssetProcessor/native/InternalBuilders/SettingsRegistryBuilder.h) | - | `engine.json`, `.setreg` |
| `.shader` | Shader Asset Builder | [Atom Gem](https://github.com/o3de/o3de/blob/development/Gems/Atom/Asset/Shader/Code/Source/Editor/AzslShaderBuilderSystemComponent.h) | [Shaders](docs/atom-guide/dev-guide/shaders) are used by the rendering system to determine how 3D geometry is rendered. | `.json`, `.azslin`, `.azshadervariant`, `.hlsl` |
| `.shadervariantlist` | Shader Variant Asset Builder | [Atom Gem](https://github.com/o3de/o3de/blob/development/Gems/Atom/Asset/Shader/Code/Source/Editor/AzslShaderBuilderSystemComponent.h) | [Shader variants](docs/atom-guide/dev-guide/shaders/azsl/shader-variant-options) are shader constants that are only used in conditional statements that can be statically optimized. | `.azshadervarianttree`, `.azshadervariant` | 

> | `.uicanvas` | UI Canvas Builder | [LyShine Gem](https://github.com/o3de/o3de/blob/development/Gems/LyShine/Code/Pipeline/LyShineBuilder/UiCanvasBuilderWorker.h) | [UI Canvas](docs/user-guide/interactivity/user-interface) files are used to define two dimensional visual elements for your project, usually used for user interfaces. | `.uicanvas` |

Author: Fixed

> | `.ent`, `.vegdescriptorlist`, and anything matching the regular expression:`(?!.*libs\/gameaudio\/).*\.xml` | XmlBuilderWorker | [LmbrCentral Gem](https://github.com/o3de/o3de/blob/development/Gems/LmbrCentral/Code/Source/Builders/CopyDependencyBuilder/XmlBuilderWorker/XmlBuilderWorker.h) | XML files are copied directly to the cache as product assets. The XML Builder exists to provide an entry point for scanning XML files for Product Dependencies, used during [Asset Bundling](docs/user-guide/packaging/asset-bundler) for packaging up content when distributing projects to end users. | `.xml` |

Author: Fixed

## Copy Jobs

> A copy job is used for assets that have no additional processing, and are copied from the source asset to the asset cache. The product asset is identical to the source asset in this case. Copy jobs are defined via settings registry, the engine default copy jobs can be found in [AssetProcessorPlatformConfig.setreg](https://github.com/o3de/o3de/blob/development/Registry/AssetProcessorPlatformConfig.setreg).

Reviewer: ```suggestion
A copy job is used for assets that have no additional processing requirements, they copy the source asset directly to the asset cache. In these cases, the product asset is identical to the source asset. Copy jobs are defined via the settings registry, the engine's default copy jobs can be found in [AssetProcessorPlatformConfig.setreg](https://github.com/o3de/o3de/blob/development/Registry/AssetProcessorPlatformConfig.setreg).
```

Author: Fixed

| `.materialtype`, `.material` | Atom Material Builder | [Atom Gem](https://github.com/o3de/o3de/blob/development/Gems/Atom/RPI/Code/Source/RPI.Builders/Material/MaterialBuilder.h) | [Materials](docs/atom-guide/look-dev/materials) contain properties to define how objects are rendered. | `.materialtype` outputs `.material` and `.azmaterialtype` products, `.material` outputs `.azmaterial` products. |
| `.resourcepool` | Atom Resource Pool Asset Builder | [Atom Gem](https://github.com/o3de/o3de/blob/development/Gems/Atom/RPI/Code/Source/RPI.Builders/ResourcePool/ResourcePoolBuilder.h) | - | `.pool`, or `.streamingimagepool` |
| `.benchmarksettings`, `.benchmark` | Benchmark Asset Worker Builder | [LmbrCentral Gem](https://github.com/o3de/o3de/blob/development/Gems/LmbrCentral/Code/Source/Builders/BenchmarkAssetBuilder/BenchmarkAssetBuilderComponent.h) | - | - |
| `.blast` | Blast Scene Builder | [Blast Gem](https://github.com/o3de/o3de/blob/development/Gems/Blast/Editor/Scripts/blast_asset_builder.py) | - | `.blast`, `.assetinfo.generated` |

> | `.cfg` | CFG Builder Worker | [LmbrCentral Gem](https://github.com/o3de/o3de/blob/development/Gems/LmbrCentral/Code/Source/Builders/CopyDependencyBuilder/CfgBuilderWorker/CfgBuilderWorker.h) | - | `.cfg` |

Reviewer 2: is this still a thing?

Author: The builder is still there and registered. I gathered this list of assets from what shows up for Automated Testing right now.

| `.pass` | Pass Asset Builder | [Atom Gem](https://github.com/o3de/o3de/blob/development/Gems/Atom/RPI/Code/Source/RPI.Builders/Pass/PassBuilder.h) | [Passes](docs/atom-guide/dev-guide/passes) are logical groupings of render work, with a defined input and output. | `.pass` |
| `.precompiledshader` | Precompiled Shader Builder | [Atom Gem](https://github.com/o3de/o3de/blob/development/Gems/Atom/Asset/Shader/Code/Source/Editor/AzslShaderBuilderSystemComponent.h) | - | `.azshader` |
| `.prefab` | Prefab Builder | [Prefab Gem](https://github.com/o3de/o3de/blob/development/Gems/Prefab/PrefabBuilder/PrefabBuilderComponent.h) | [Prefabs](docs/user-guide/interactivity/prefabs) are a foundational element for building an O3DE project, a prefab is a collection of entities. | `.spawnable` |
| `.ts` | Qt Translation File Builder | [LmbrCentral Gem](https://github.com/o3de/o3de/blob/development/Gems/LmbrCentral/Code/Source/Builders/TranslationBuilder/TranslationBuilderComponent.h) | Qt translation files. Qt is only enabled in tools, so these can be useful for translating tools to different languages, but not projects. | `.qm` |

> | `.fbx`, `.glb`, `gltf`, `stl` | Scene Builder | [SceneProcessing Gem](https://github.com/o3de/o3de/blob/development/Gems/SceneProcessing/Code/Source/SceneBuilder/SceneBuilderWorker.h) | [Scene files](docs/user-guide/assets/scene-settings/scene-format-support) are used to define 3D content, such as meshes and animations. | `.motion`, `.procprefab`, `.azmaterial`, `.azbuffer`, `.azlod`, `.azmodel`, `.blast_chunks`, `.pxmesh`, `.skinmeta`, `.actor` |

Reviewer 2: blast_chunks is a product asset output from the Blast gem

Author: The Asset Processor says it's from the Scene Builder Worker, and not the Blast Gem.

| `engine.json` | Settings Registry Builder | [Asset Processor](https://github.com/o3de/o3de/blob/development/Code/Tools/AssetProcessor/native/InternalBuilders/SettingsRegistryBuilder.h) | - | `engine.json`, `.setreg` |
| `.shader` | Shader Asset Builder | [Atom Gem](https://github.com/o3de/o3de/blob/development/Gems/Atom/Asset/Shader/Code/Source/Editor/AzslShaderBuilderSystemComponent.h) | [Shaders](docs/atom-guide/dev-guide/shaders) are used by the rendering system to determine how 3D geometry is rendered. | `.json`, `.azslin`, `.azshadervariant`, `.hlsl` |
| `.shadervariantlist` | Shader Variant Asset Builder | [Atom Gem](https://github.com/o3de/o3de/blob/development/Gems/Atom/Asset/Shader/Code/Source/Editor/AzslShaderBuilderSystemComponent.h) | [Shader variants](docs/atom-guide/dev-guide/shaders/azsl/shader-variant-options) are shader constants that are only used in conditional statements that can be statically optimized. | `.azshadervarianttree`, `.azshadervariant` | 

> | `.slice` | Slice Builder | [Lmbr Central Gem](https://github.com/o3de/o3de/blob/development/Gems/LmbrCentral/Code/Source/Builders/SliceBuilder/SliceBuilderWorker.h) | Slices are a legacy system for storing groups of entities in assets. The slice system is being replaced by the Prefab system. | `.slice`, `.dynamicslice` |

Reviewer 2: I think we should officially deprecate "slice" files by not documenting them.
